cmd/soroban-rpc/internal/events/events.go

```diff
@@ -243,11 +196,14 @@ func readEvents(txReader *ingest.LedgerTransactionReader) ([]event, error) {
 			}
 		}
 	}
+	if err := txReader.Close(); err != nil {
```

Comment:
could you defer that to ensure that we call Close in the case of an error ?
( the returned error in this case is a bit tricky )